projects/storefrontlib/src/cms-components/product/product-interests/stock-notification/notification-dialog/notification-dialog.component.spec.ts

```diff
@@ -59,61 +59,53 @@ describe('NotificationDialogComponent', () => {
   beforeEach(() => {
     fixture = TestBed.createComponent(NotificationDialogComponent);
     component = fixture.componentInstance;
+    component.subscribeSuccess$ = of(true);
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
 
-  it('should show selected channels when selectedChannels are exist', () => {
-    selectedChannels = [
-      {
-        channel: 'EMAIL',
-        value: 'test@sap.com',
-      },
-      {
-        channel: 'SMS',
-        value: '13800000831',
-      },
-    ];
-    component.subscribeSuccess$ = of(true);
-    component.selectedChannels = selectedChannels;
+  it('should show notification dialog', () => {
     fixture.detectChanges();
-    const plist = fixture.debugElement.queryAll(By.css('p'));
-    expect(plist.length).toEqual(6);
-    expect(plist[1].nativeElement.textContent).toContain('EMAIL: test@sap.com');
-    expect(plist[2].nativeElement.textContent).toContain('SMS: 13800000831');
+
+    expect(el.query(By.css('[data-test]="notifydialog-title"'))).toBeTruthy();
+    expect(el.query(By.css('[data-test]="close-button"'))).toBeTruthy();
+    expect(el.query(By.css('[data-test]="notify-channel"'))).toBeTruthy();
+    expect(
+      el.queryAll(By.css('[data-test]="notifydialog-link"')).length
+    ).toEqual(2);
+    expect(el.query(By.css('[data-test]="ok-button"'))).toBeTruthy();
   });
 
-  it('should show manage channels link and manage subscriptions link', () => {
-    component.subscribeSuccess$ = of(true);
+  it('should show spinner when loading', () => {
+    component.subscribeSuccess$ = of(false);
+    productInterestService.createBackInStock.and.returnValue({});
     fixture.detectChanges();
-    const alist = fixture.debugElement.queryAll(By.css('a'));
-    expect(alist[0].nativeElement.textContent).toContain(
-      'stockNotification.subscriptionDialog.manageChannelsLink'
-    );
-    expect(alist[1].nativeElement.textContent).toContain(
-      'stockNotification.subscriptionDialog.manageSubscriptionsLink'
-    );
+
+    expect(el.query(By.css('.cx-spinner'))).toBeTruthy();
   });
 
-  it('should show loading template when data is not ready', () => {
-    selectedChannels = [];
-    component.selectedChannels = selectedChannels;
-    component.subscribeSuccess$ = of(false);
+  it('should be able to close dialog', () => {
     fixture.detectChanges();
-    const notificationPreference = fixture.debugElement.queryAll(
-      By.css('.cx-notification-preference-span')
-    );
-    expect(notificationPreference.length).toEqual(0);
+
+    el.query(By.css('[data-test]="close-button"')).nativeElement.click();
+    expect(ngbActiveModal.dismiss).toHaveBeenCalled();
   });
 
   it('should be able to close dialog', () => {
```

Comment:
The name of this case is the same as before, please merge them or provide different name